downloads/swan-lake-release-notes/swan-lake-2201.7.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,243 @@
+
+  This may result in compilation errors that were previously not identified.
+
+    ```ballerina
+    const int CONST = 1 + 7;
+    CONST num = 2; // Compilation error now.
+    ```
+  This results in a compilation error now since `num`, which is of type `CONST`, can only be integer `8`.
+  
+- A bug that resulted in compilation errors not being logged for duplicate keys via computed name fields of a mapping constructor in a mapping constant declaration has been fixed. This previously resulted in a panic at runtime instead.
+
+    ```ballerina
+    const MESSAGE = "message";
+    const map<string> HTTP_OK = {httpCode: "200", message: "OK", [MESSAGE] : "BAD REQUEST"}; // Compilation error now.
+    ```
+
+- Fixed a bug in the configurable TOML syntax validation for the module structure. The error message for an invalid TOML module structure is now improved to provide the variable name.
+
+    For example, consider a non-default module named `foo.bar`, which contains the following configurable variables.
+
+    ```ballerina
+    configurable int intVar = ?;
+    configurable string stringVar = ?;
+    ```
+
+    If an invalid TOML structure is found for the `foo.bar` module in the `Config.toml` file, it results in the following error messages.
+
+    ```
+    [Config.toml:(1:1,2:23)] invalid TOML structure found for module 'foo.bar' with variable 'intVar'. Please provide the module name as '[foo.bar]'
+    [Config.toml:(1:1,2:23)] invalid TOML structure found for module 'foo.bar' with variable 'stringVar'. Please provide the module name as '[foo.bar]'
+    ```
+
+## Platform updates
+
+### Official support for generating GraalVM native executables
+
+Ballerina now officially supports generating GraalVM native executables and the language and the standard libraries are compatible with the GraalVM native executable generation. In addition to that, GraalVM incompatibility warnings will be printed for any incompatible modules in the application. To explore more on this support, see [Build a GraalVM executable](/learn/graalvm-executable-overview/).
+
+## Language updates
+
+### New features
+
+#### Introduction of the `group by` and `collect` clauses
+
+The language now supports the `group by` and `collect` clauses to perform aggregation-related operations.
+
+The `group by` clause is used to group a collection based on a grouping-key as shown in the example below. The grouping-key will be unique to each group.
+
+```ballerina
+import ballerina/io;
+
+type Order record {|
+    string name;
+    int price;
+|};
+
+public function main() {
+    Order[] orders = [{name: "Item1", price: 91}, {name: "Item2", price: 83}, {name: "Item1", price: 75}, {name: "Item2", price: 88}];
+    var averages = from var {name, price} in orders
+        group by name
+        select {name, avg: avg(price)};
+}
+```
+
+The `collect` clause is used to group a collection into one group as shown in the example below.
```

Comment:
Is " as shown in the example below" really required?

```suggestion
The `collect` clause is used to group a collection into one group.
```

Reply:
There is no harm adding it to connect the description with the code below.